pkg/sql/sem/tree/datum.go

```diff
@@ -273,6 +274,11 @@ func (*DBool) ResolvedType() types.T {
 	return types.Bool
 }
 
+// AsJSON implements transformation to JSON
```

Comment:
Two style nits: we usually put periods after comments, and also we usually write these comments in the form `AsJSON implements the json.JSONizable interface`.

In addition, the specificity here is nice, but in the interest of keeping the interface of the `json` package smaller where possible, maybe it would be easier to just use `json.MakeJSON` here and in other methods where possible? It will be a little slower because of the switch, I think, but that's something that can be changed if we run into problems with it later. You might be able to even just add switch cases for things like `time` and `timeofday`?